shared/src/writeToTypegenFile.ts

```diff
@@ -0,0 +1,32 @@
+import * as fs from "fs";
+import * as prettier from "prettier";
+import { promisify } from "util";
+import { getTypegenOutput } from ".";
+
+export const writeToTypegenFile = async (opts: {
+  filePath: string;
+  event: Parameters<typeof getTypegenOutput>[0];
```

Comment:
q: any particular reason why this is called an event here? i looks like a pretty imperative call 🤔

Reply:
Yeah it's just called an event elsewhere in the code (`XStateUpdateEvent` from VSCode) so I kept it here.